python_modules/libraries/dagster-dg/docs/app/packages/[pkg]/[component]/page.tsx

```diff
@@ -0,0 +1,57 @@
+import ComponentDetails from '@/app/components/ComponentDetails';
+import getContents, {getComponent} from '@/util/getContents';
+
+import styles from './ComponentPage.module.css';
+
+interface Props {
+  params: Promise<{pkg: string; component: string}>;
+}
+
+export default async function ComponentPage({params}: Props) {
+  const {pkg, component} = await params;
+  const contents = await getContents();
```

Comment:
I think we shouldn't be calling this inside the component. We should be relying on `getStaticProps`

Reply:
`getStaticProps` is not used with App router.

It's possible that there's something I should be doing a bit differently here, but I'm trying to roughly follow https://nextjs.org/docs/app/building-your-application/data-fetching/fetching.

Reply:
Ah yeah ignore me.  Looks like with app-router you're expected to follow this pattern of using a server-component to pass the data to the client components. We just need to make sure this page is never client rendered (which I am not sure if its possible or whether nextjs will complain, one way to test would be to add a "use client" directive at the top of this file to see what happens. If next.js allows the page to be rendered on the client then I would suggest adding a "server only" directive to make sure that we don't accidentally pull all of the JSON on the client.

Reply:
Confirmed that when I add a `use client` to this page, the build breaks because the page includes a `generateStaticParams` function.

The root `Page` component does not use `generateStaticParams`, but when that is given the `use client` directive, it results in this error in the browser:

```
error-boundary-callbacks.ts:101 Uncaught Error: An unknown Component is an async Client Component. Only Server Components can be async at the moment. This error is often caused by accidentally adding `'use client'` to a module that was originally written for the server.
```

So I think things look like they're behaving as expected. ☺️